instrumentation/runtime/producer.go

```diff
@@ -0,0 +1,115 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package runtime // import "go.opentelemetry.io/contrib/instrumentation/runtime"
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"runtime/metrics"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+var startTime time.Time
+
+func init() {
+	startTime = time.Now()
+}
+
+var histogramMetrics = []string{goSchedLatencies}
+
+// Producer is a metric.Producer, which provides precomputed histogram metrics from the go runtime.
+type Producer struct {
+	collector *goCollector
+}
+
+var _ metric.Producer = (*Producer)(nil)
+
+// NewProducer creates a Producer which provides precomputed histogram metrics from the go runtime.
+func NewProducer(opts ...ProducerOption) *Producer {
+	c := newProducerConfig(opts...)
+	return &Producer{
+		collector: newCollector(c.MinimumReadMemStatsInterval, histogramMetrics),
+	}
+}
+
+// Produce returns precomputed histogram metrics from the go runtime, or an error if unsuccessful.
+func (p *Producer) Produce(context.Context) ([]metricdata.ScopeMetrics, error) {
+	p.collector.refresh()
+	// Use the last collection time (which may or may not be now) for the timestamp.
+	histDp := convertRuntimeHistogram(p.collector.getHistogram(goSchedLatencies), p.collector.lastCollect)
+	if len(histDp) == 0 {
+		return nil, fmt.Errorf("unable to obtain go.schedule.duration metric from the runtime")
+	}
+	return []metricdata.ScopeMetrics{
+		{
+			Scope: instrumentation.Scope{
```

Comment:
note that this is a separate instrumentation scope from the one that will be produced by runtime instrumentation.  The OTel SDK does not merge those today, but I'm not sure how problematic that will be.  I propose we merge this as-is, and then see what impact it has, if any.